### PR TITLE
feat: Support for proxy server settings in ~/.m2/settings.xml

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk-maven-plugin/blob/master/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Fix for #24, in supporting of proxy set through `~/.m2/settings.xml`
#### How should this be manually tested?
Yes
 #### Any background context you want to provide?
https://github.com/snyk/snyk-maven-plugin/pull/40
#### What are the relevant tickets?
https://github.com/snyk/snyk-maven-plugin/issues/24

### CODE REVIEW

**Comment by adrukh** [comment]
@lili2311 I approved too soon, please promote plugin version in `pom.xml`

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
   <groupId>io.snyk</groupId>
   <artifactId>snyk-maven-plugin</artifactId>
-  <version>1.2.2</version>
+  <version>1.3.2</version>
```
Comment (yuliabaron) [inline pom.xml]: shouldn't you change the version in the test pom.xml files as well?

Comment (adrukh) [reply pom.xml]: Also, this should be either `1.2.3` or `1.3.0`? Not `1.3.2`?

Comment (lili2311) [reply pom.xml]: yep sorry wrong place

Comment (lili2311) [reply pom.xml]: Updated

**Pull request merged**